test/jdk/java/nio/channels/SocketChannel/OpenLeak.java

```diff
@@ -26,27 +26,77 @@
  * @summary SocketChannel.open(SocketAddress) leaks file descriptor if
  *     connection cannot be established
  * @build OpenLeak
- * @run main/othervm -Djava.security.manager=allow OpenLeak
+ * @run junit/othervm OpenLeak
  */
 
+import java.net.ConnectException;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.net.SocketAddress;
 import java.nio.channels.SocketChannel;
+import java.nio.channels.UnresolvedAddressException;
+
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
 
 public class OpenLeak {
 
-    public static void main(String[] args) throws Exception {
-        InetAddress lh = InetAddress.getLocalHost();
-        InetSocketAddress isa = new InetSocketAddress(lh, 12345);
+    private static final int MAX_LOOP = 250000;
 
-        System.setSecurityManager( new SecurityManager() );
-        for (int i=0; i<100000; i++) {
-            try {
-                SocketChannel.open(isa);
-                throw new RuntimeException("This should not happen");
-            } catch (SecurityException x) { }
-        }
+    @Test
+    public void test() throws Exception {
+        InetAddress lo = InetAddress.getLoopbackAddress();
+
+        // Try to find a suitable port that will cause a
+        // Connection Rejected exception
+        // port 47 is reserved - there should be nothing there...
+        InetSocketAddress isa = new InetSocketAddress(lo, 47);
+        try (SocketChannel sc1 = SocketChannel.open(isa)) {
+            // If we manage to connect, let's try to use some other
+            // port.
+            // port 51 is reserved too - there should be nothing there...
+            isa = new InetSocketAddress(lo, 51);
+            try (SocketChannel sc2 = SocketChannel.open(isa)) {};
+            // OK, last attempt...
+            // port 61 is reserved too - there should be nothing there...
+            isa = new InetSocketAddress(lo, 61);
+            try (SocketChannel sc3 = SocketChannel.open(isa)) {};
+            Assumptions.abort("Could not find a suitable port");
+            return;
+        } catch (ConnectException x) { }
 
+        // create an unresolved address to test another path
+        //   where close should be called
+        SocketAddress sa = InetSocketAddress.createUnresolved(isa.getHostString(), isa.getPort());
+
+        System.out.println("Expecting Connection Refused for " + isa);
+        System.out.println("Expecting UnresolvedAddressException for " + sa);
```

Comment:
JUnit prints the STARTED/SUCCESSFUL/etc messages to System.err so you probably want the trace messages to go to the same stream.